cmd/influx/main.go

```diff
@@ -53,8 +55,14 @@ func main() {
 	fs.StringVar(&c.Password, "password", c.Password, `password to connect to the server.  Leaving blank will prompt for password (--password="")`)
 	fs.StringVar(&c.Database, "database", c.Database, "database to connect to the server.")
 	fs.StringVar(&c.Format, "output", default_format, "format specifies the format of the server responses:  json, csv, or column")
+	fs.BoolVar(&c.Dump, "dump", false, "dump the contents of the given database to stdout")
 	fs.Parse(os.Args[1:])
 
+	if c.Dump {
+		c.connect("")
+		c.dump()
```

Comment:
No need to call it at all.  As long as we just `return` and don't do an `os.Exit` or anything like that, the defer will pick up as soon as we go out of scope, which is what we want.